service_configuration_lib/spark_config.py

```diff
@@ -66,6 +73,12 @@ def get_mesos_spark_env(
 ) -> Mapping[str, str]:
     _check_non_configurable_spark_opts(user_spark_opts)
 
+    for volume in volumes:
+        volume_elements = volume.split(':')
+        if len(volume_elements) == 3 and volume_elements[-1] not in ('rw', 'ro'):
```

Comment:
I vote that we call `.lower()` here; this way users don't have to worry about casing.  But it's still worth having the error in case of a typo.